File: 566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        n,m=len(mat),len(mat[0])
        if r*c!=n*m:
            return mat
        x,y=0,0
        res=[]
        for i in range(r):
            k=[]
            for j in range(c):
                k.append(mat[x][y])
                y+=1
                if y==m:
                    y=0
                    x+=1
            res.append(k)
        return res

File: 566. Reshape the Matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        if(r*c!=mat.length*mat[0].length){
            return mat;
        }
        int r1,c1;
        r1=0;
        c1=0;
        int[][] transform =new int[r][c];
        for(int i=0;i<r;i++){
            for(int j=0;j<c;j++){
                transform[i][j]=mat[r1][c1];
                if(c1==mat[0].length-1){
                    r1++;
                    c1=0;
                    continue;
                }
                c1++;
            }
        }
        return transform;
    }
}
